Readme.md

```diff
@@ -146,6 +146,37 @@ Start your file with a module docstring. Do not put a shebang line (`#!/usr/bin/
 
 > *Rationale:* a shebang line is useless for non-executable files. An `__author__` line just gets out of date, and is better determined by looking at source control history in any case. Code is automatically copyrighted; a copyright line doesn't help. No need to put this useless boilerplate at the top of the file!
 
+## Python Tests
+
+There are a few conventions to follow while writing unit tests. Tests for a particular method or function should be prefixed with 'test_' with underscores separating words. This agrees with the convention used in the [python unittest documentation](https://docs.python.org/2/library/unittest.html) and also will be compatible with the [discovery mechanism used by pytest](http://pythontesting.net/framework/pytest/pytest-introduction/#discovery) and other tools.
+
+Unlike method or function names, it is [recommended to use long and descriptive](http://docs.python-guide.org/en/latest/writing/tests/) names for your test methods. This allows them to be easily identified in command line output when tests fail. When naming your test method, use the following structure:
+
+```
+test_*<item_under_test>*_when_*<condition>*_expect_*<outcome>*.
```

Comment:
Agree on if/then. 

I'd argue for underscores for the test function names for some of the reasons mentioned above. This is a place where PEP8 is a little self-contradicting in that even though it recommends brevity in line columns, it also says use underscores in function names wherever possible (https://www.python.org/dev/peps/pep-0008/#id36). Personally, I find underscores a lot easier to read/parse. If our name is getting so long that we're trying to save 5-10 characters by eliminating underscores, then I think we probably just want to shorten the name at that point.